src/components/Trade/hooks/useSwapper/useSwapper.tsx

```diff
@@ -149,79 +108,16 @@ export const useSwapper = () => {
     return txid
   }, [activeSwapper, isExactAllowance, activeQuote, wallet])
 
-  const getTrade = useCallback(async () => {
-    if (!sellAsset) throw new Error('No sellAsset')
-    if (!activeSwapper) throw new Error('No swapper available')
-    if (!sellAmountCryptoPrecision) throw new Error('Missing sellTradeAsset.amount')
-    if (!sellAsset) throw new Error('Missing sellAsset')
-    if (!buyAsset) throw new Error('Missing buyAsset')
-    if (!wallet) throw new Error('Missing wallet')
-    if (!receiveAddress) throw new Error('Missing receiveAddress')
-    if (!sellAssetAccountId) throw new Error('Missing sellAssetAccountId')
-    if (!sellAccountBip44Params) throw new Error('Missing sellAccountBip44Params')
-    if (!buyAccountBip44Params) throw new Error('Missing buyAccountBip44Params')
-    if (!sellAccountMetadata) throw new Error('Missing sellAccountMetadata')
-
-    const buildTradeCommonArgs: BuildTradeInputCommonArgs = {
-      sellAmountBeforeFeesCryptoBaseUnit: toBaseUnit(
-        sellAmountCryptoPrecision,
-        sellAsset.precision,
-      ),
-      sellAsset,
-      buyAsset,
-      wallet,
-      sendMax: isSendMax,
-      receiveAddress,
-      slippage,
-    }
-    const sellAssetChainId = sellAsset.chainId
-    if (isSupportedCosmosSdkSwappingChain(sellAssetChainId)) {
-      const { accountNumber } = sellAccountBip44Params
-      const { accountNumber: receiveAccountNumber } = buyAccountBip44Params
-      return activeSwapper.buildTrade({
-        ...buildTradeCommonArgs,
-        chainId: sellAssetChainId,
-        accountNumber,
-        receiveAccountNumber,
-      })
-    } else if (isSupportedNonUtxoSwappingChain(sellAssetChainId)) {
-      const { accountNumber } = sellAccountBip44Params
-      return activeSwapper.buildTrade({
-        ...buildTradeCommonArgs,
-        chainId: sellAssetChainId,
-        accountNumber,
-      })
-    } else if (isSupportedUtxoSwappingChain(sellAssetChainId)) {
-      const { accountType, bip44Params } = sellAccountMetadata
-      const { accountNumber } = bip44Params
-      if (!bip44Params) throw new Error('no bip44Params')
-      if (!accountType) throw new Error('no accountType')
-      const sellAssetChainAdapter = getChainAdapterManager().get(
-        sellAssetChainId,
-      ) as unknown as UtxoBaseAdapter<UtxoChainId>
-      const { xpub } = await sellAssetChainAdapter.getPublicKey(wallet, accountNumber, accountType)
-      return activeSwapper.buildTrade({
-        ...buildTradeCommonArgs,
-        chainId: sellAssetChainId,
-        accountNumber,
-        accountType,
-        xpub,
-      })
-    }
-  }, [
-    sellAsset,
-    activeSwapper,
-    sellAmountCryptoPrecision,
-    buyAsset,
-    wallet,
-    receiveAddress,
-    sellAssetAccountId,
-    sellAccountBip44Params,
-    buyAccountBip44Params,
-    sellAccountMetadata,
-    isSendMax,
-    slippage,
-  ])
+  const getTrade = useCallback(
+    async () =>
+      await getTradeForWallet({
+        wallet,
```

Comment:
We may want to make this guy required in `SelectGetTradeForWalletArgs` vs. `HDWallet | null` to keep it consistent with other diffed components/hooks in this diff throwing `if (!wallet)`

Reply:
https://github.com/shapeshift/web/pull/4078/commits/5cb27017be7955259ff325dfd4e1bce1e18a58cc 👌